src/Microsoft.AspNetCore.DataProtection/SimpleActivator.cs

```diff
@@ -52,5 +54,33 @@ public virtual object CreateInstance(Type expectedBaseType, string implementatio
             // This will throw if the ctor cannot be called.
             return Activator.CreateInstance(implementationType);
         }
+
+        private static string RemoveVersionFromAssemblyName(string typeName)
+        {
+            var idx = typeName.IndexOf(',');
+            if (idx <= 0 || (idx + 1) >= typeName.Length)
+            {
+                return typeName;
+            }
+
+            // type name includes assembly name
+            var assemblyName = typeName.Substring(idx + 1);
```

Comment:
Should it be a specific list or is `AssemblyName.StartsWith("Microsoft.AspNetCore")` ok?